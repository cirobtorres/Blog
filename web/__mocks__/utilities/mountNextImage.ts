const nextOptimizedImage = (filePath: string) => {
  if (filePath) {
    const filename = filePath.substring(
      filePath.lastIndexOf("/") + 1,
      filePath.lastIndexOf(".")
    );
    const extension = filePath.substring(filePath.lastIndexOf(".") + 1);
    const path = filePath.substring(0, filePath.lastIndexOf("/") + 1);
    const URL = `${path}${filename}.${extension}`;
    const encodedURL = encodeURIComponent(URL);
    const width = "w=3840";
    const quality = "q=75";
    const nextOptimizedImage = `/_next/image?url=${encodedURL}&${width}&${quality}`;
    return nextOptimizedImage;
  }
  return null;
};

interface ToEncodeURIAttributes {
  imageFilename: string;
  imageFileExtension: string;
  path: string;
  width: number;
}

function mountLocalImage({
  imageFilename,
  imageFileExtension,
  path,
  width,
}: ToEncodeURIAttributes) {
  // Mock local images
  const quality = 75;
  const URL = `${path}${imageFilename}.${imageFileExtension}`;
  const encodedURL = encodeURIComponent(URL);
  const nextOptimizedImage = `/_next/image?url=${encodedURL}&w=${width}&q=${quality}`;
  return nextOptimizedImage;
}

function requestBackEndImage(
  filePath?: string,
  width: number = 1920,
  options?: {
    PROTOCOL?: string;
    IP?: string;
    PORT?: string;
  }
) {
  // Mock images managed by Strapi
  const quality = 75;

  if (filePath) {
    const filename = filePath.substring(
      filePath.lastIndexOf("/") + 1,
      filePath.lastIndexOf(".")
    );
    const extension = filePath.substring(filePath.lastIndexOf(".") + 1);
    const path = filePath.substring(0, filePath.lastIndexOf("/") + 1);

    const protocol = options ? options.PROTOCOL : "http";
    const ip = options ? options.IP : "127.0.0.1";
    const port = options ? options.PORT : "1337";
    const backendURL = `${protocol}://${ip}:${port}`;

    const URL = `${backendURL}${path}${filename}.${extension}`;

    const encodedURL = encodeURIComponent(URL);
    const nextOptimizedImage = `/_next/image?url=${encodedURL}&w=${
      width * 2
    }&q=${quality}`;

    return nextOptimizedImage;
  }

  const URL = "/images/not-authenticated.png";
  const encodedURL = encodeURIComponent(URL);

  const nextOptimizedImage = `/_next/image?url=${encodedURL}&w=${
    width * 2
  }&q=${quality}`;

  return nextOptimizedImage;
}

export { nextOptimizedImage, mountLocalImage, requestBackEndImage };
