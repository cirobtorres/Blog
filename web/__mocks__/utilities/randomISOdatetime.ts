import { faker } from "@faker-js/faker";

export function generateRandomCreatedAt() {
  const createdAt = faker.date.past().toISOString();
  const updatedAt = createdAt;
  const publishedAt = createdAt;
  return {
    createdAt,
    updatedAt,
    publishedAt,
  };
}
