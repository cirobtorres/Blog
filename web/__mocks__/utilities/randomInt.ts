import { faker } from "@faker-js/faker";

export const randomInt = (min: number, max: number) =>
  faker.number.int({ min, max });
