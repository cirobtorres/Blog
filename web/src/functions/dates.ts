function formatDateToCustomFormat(ISOdate: string) {
  const date = new Date(ISOdate);

  const months = [
    "janeiro",
    "fevereiro",
    "março",
    "abril",
    "maio",
    "junho",
    "julho",
    "agosto",
    "setembro",
    "outubro",
    "novembro",
    "dezembro",
  ];

  const day = date.getDate();
  const month = months[date.getMonth()];
  const year = date.getFullYear();
  const hours = String(date.getHours()).padStart(2, "0");
  const minutes = String(date.getMinutes()).padStart(2, "0");

  return `${day} de ${month} de ${year}, às ${hours}:${minutes}`;
}

export { formatDateToCustomFormat };
