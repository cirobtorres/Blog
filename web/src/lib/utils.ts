import { remark } from "remark";
import html from "remark-html";
import { clsx, type ClassValue } from "clsx";
import { twMerge } from "tailwind-merge";

const cn = (...inputs: ClassValue[]) => {
  return twMerge(clsx(inputs));
};

const convertMarkdowToHtmlString = async (rawHtml: string | undefined) => {
  const processedContent = await remark().use(html).process(rawHtml);
  const stringHtml = processedContent.toString();
  return stringHtml;
};

export { cn, convertMarkdowToHtmlString };
