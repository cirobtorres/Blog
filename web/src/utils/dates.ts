function formatDateToCustomFormat(ISOdate: string) {
  const date = new Date(ISOdate);

  const months = [
    "janeiro",
    "fevereiro",
    "março",
    "abril",
    "maio",
    "junho",
    "julho",
    "agosto",
    "setembro",
    "outubro",
    "novembro",
    "dezembro",
  ];

  const day = date.getDate();
  const month = months[date.getMonth()];
  const year = date.getFullYear();
  const hours = String(date.getHours()).padStart(2, "0");
  const minutes = String(date.getMinutes()).padStart(2, "0");

  return `${day} de ${month} de ${year}, às ${hours}:${minutes}`;
}

function formatDateToYouTubeLikeFormat(ISOdate: string) {
  const date = new Date(ISOdate);
  const now = new Date();
  const diffInMs = now.getTime() - date.getTime();
  const diffInSeconds = Math.floor(diffInMs / 1000);
  const diffInMinutes = Math.floor(diffInSeconds / 60);
  const diffInHours = Math.floor(diffInMinutes / 60);
  const diffInDays = Math.floor(diffInHours / 24);
  const diffInMonths = Math.floor(diffInDays / 30);

  if (diffInSeconds < 60) {
    return `${diffInSeconds <= 1 ? "agora" : `há ${diffInSeconds} segundos`}`;
  }

  if (diffInMinutes < 60) {
    return `há ${diffInMinutes} ${diffInMinutes === 1 ? "minuto" : "minutos"}`;
  }

  if (diffInHours < 24) {
    return `há ${diffInHours} ${diffInHours === 1 ? "hora" : "horas"}`;
  }

  if (diffInDays < 30) {
    return `há ${diffInDays} ${diffInDays === 1 ? "dia" : "dias"}`;
  }

  return `há ${diffInMonths} ${diffInMonths === 1 ? "mês" : "meses"}`;
}

export { formatDateToCustomFormat, formatDateToYouTubeLikeFormat };
