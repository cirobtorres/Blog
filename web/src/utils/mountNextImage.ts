interface ToEncodeURIAttributes {
  imageFilename: string;
  imageFileExtension: string;
  path: string;
  width: number;
  quality: number;
}

function mountNextImage({
  imageFilename,
  imageFileExtension,
  path,
  width,
  quality,
}: ToEncodeURIAttributes) {
  const URL = `${path}${imageFilename}.${imageFileExtension}`;
  const encodedURL = encodeURIComponent(URL);
  const nextOptimizedImage = `/_next/image?url=${encodedURL}&w=${width}&q=${quality}`;
  return nextOptimizedImage;
}

function requestBackEndNextImage(
  imageFilename: string,
  imageFileExtension: string,
  path: string,
  width: number,
  quality: number,
  options?: {
    PROTOCOL?: string;
    IP?: string;
    PORT?: string;
  }
) {
  const URL = `${options ? options.PROTOCOL : "http"}://${
    options ? options.IP : "127.0.0.1"
  }:${
    options ? options.PORT : "1337"
  }${path}${imageFilename}.${imageFileExtension}`;
  const encodedURL = encodeURIComponent(URL);
  const nextOptimizedImage = `/_next/image?url=${encodedURL}&w=${width}&q=${quality}`;

  return nextOptimizedImage;
}

export { mountNextImage, requestBackEndNextImage };
