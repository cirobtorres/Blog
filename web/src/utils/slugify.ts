const slugify = (str: string) => {
  return str
    .toLowerCase()
    .normalize("NFD") // Normaliza a string para decompor caracteres acentuados
    .replace(/[\u0300-\u036f]/g, "") // Remove marcas diacríticas (acentos)
    .trim() // Remove espaços extras no início e no final
    .replace(/[^\w\s-]/g, "") // Remove caracteres especiais
    .replace(/[\s_-]+/g, "-") // Substitui espaços e underscores por hifens
    .replace(/^-+|-+$/g, "");
};

export default slugify;
