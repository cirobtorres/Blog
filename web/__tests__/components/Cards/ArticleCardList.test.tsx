import { faker } from "@faker-js/faker";
import { act, render, screen } from "@testing-library/react";
import {
  ArticleCard,
  ArticleCardList,
} from "../../../src/components/Cards/ArticleCardList";
import { requestBackEndImage } from "../../../__mocks__/utilities/mountNextImage";
import { formatDateToCustomFormat } from "@/utils/dates";
import { axe, toHaveNoViolations } from "jest-axe";
import slugify from "@/utils/slugify";

faker.seed(1); // Snapshots

const randomInt = (max = 10, min = 1) => {
  return faker.number.int({ min, max });
};

const title = faker.lorem.sentence(randomInt());
const slug = slugify(faker.lorem.text());
// const datetime = String(faker.date.anytime()); // Inconsistent
const datetime = "2025-03-10T23:11:17.381Z";
const url =
  faker.system.directoryPath() + "/" + faker.system.commonFileName("png");

const mockedArticleList = Array.from({ length: randomInt() }).map(() => {
  return {
    documentId: faker.string.uuid(),
    title: title,
    slug: slug,
    description: faker.lorem.paragraph(),
    createdAt: datetime,
    updatedAt: datetime,
    publishedAt: datetime,
    cover: {
      documentId: faker.string.uuid(),
      url: url,
      alternativeText: faker.lorem.sentence(randomInt(12)),
      caption: faker.lorem.sentence(randomInt(20)),
      width: 1920,
      height: 1080,
    },
  };
});

describe("ArticleCardList", () => {
  it("is in the document", () => {
    render(<ArticleCardList articles={mockedArticleList} />);
    const list = screen.getByTestId("article-card-list");
    expect(list).toBeInTheDocument();
  });

  it("is NOT in the document (article list is empty)", () => {
    render(<ArticleCardList articles={[]} />);
    const list = screen.queryByTestId("article-card-list");
    expect(list).not.toBeInTheDocument();
  });

  it("should have no accessibility violations", async () => {
    const { container } = render(
      <ArticleCardList articles={mockedArticleList} />
    );
    await act(async () => {
      const results = await axe(container);
      expect.extend(toHaveNoViolations);
      expect(results).toHaveNoViolations();
    });
  });

  // TODO: test article with no image (no src and no alt)

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <ArticleCardList articles={mockedArticleList} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("ArticleCard", () => {
  it("is in the document", () => {
    render(<ArticleCard article={mockedArticleList[0]} />);
    const link = screen.getByTestId("article-card-link");
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute(
      "href",
      `/artigos/${mockedArticleList[0].documentId}/${mockedArticleList[0].slug}`
    );
  });

  it("renders an image", () => {
    render(<ArticleCard article={mockedArticleList[0]} />);

    const imageContainer = screen.getByTestId("article-card-image");
    const image = imageContainer.querySelector("img");

    const nextJsImage = requestBackEndImage(mockedArticleList[0].cover.url);

    const alt = mockedArticleList[0].cover.alternativeText;

    expect(imageContainer).toHaveClass("relative h-[200px] shrink-0");
    expect(image).toHaveClass("absolute object-cover");
    expect(image).toHaveAttribute("src", nextJsImage);
    expect(image).toHaveAttribute("alt", alt);
  });

  it("renders title, description and datetime correctly", () => {
    render(<ArticleCard article={mockedArticleList[0]} />);

    const title = screen.getByTestId("article-card-title");
    const description = screen.getByTestId("article-card-description");
    const datetime = screen.getByTestId("article-card-datetime");

    const renderedDatetime = formatDateToCustomFormat(
      mockedArticleList[0].createdAt
    );

    expect(title).toHaveTextContent(mockedArticleList[0].title);
    expect(title).toHaveClass("max-h-[calc(1.5rem_*_3)] line-clamp-3");
    expect(description).toHaveTextContent(mockedArticleList[0].description);
    expect(description).toHaveClass("max-h-[calc(1.25rem_*_3)] line-clamp-3");
    expect(datetime).toHaveTextContent(renderedDatetime);
  });

  it("should have no accessibility violations", async () => {
    const { container } = render(
      <ArticleCard article={mockedArticleList[0]} />
    );
    await act(async () => {
      const results = await axe(container);
      expect.extend(toHaveNoViolations);
      expect(results).toHaveNoViolations();
    });
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <ArticleCard article={mockedArticleList[0]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
