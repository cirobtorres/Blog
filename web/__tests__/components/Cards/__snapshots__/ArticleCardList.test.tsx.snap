// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard matches the snapshot 1`] = `
<DocumentFragment>
  <a
    aria-label="Clique para ler mais sobre o artigo"
    class="border border-blog-border overflow-hidden h-full w-full flex flex-col rounded focus-visible:outline focus-visible:outline-2 focus-visible:outline-blog-foreground-readable-hover"
    data-testid="article-card-link"
    href="/artigos/bee102e1-6f8b-45ad-90cf-b4c17e4420a3/crux-fugit-curatio-stillicidium-bardus"
    id="article-card-link"
    role="link"
  >
    <div
      class="relative h-[200px] shrink-0 mb-2"
      data-testid="article-card-image"
    >
      <img
        alt="Approbo vicissitudo subiungo."
        class="absolute object-cover"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        role="img"
        sizes="(max-width: 1920) 100vw, (max-width: 960) 50vw, 33vw"
        src="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=3840&q=75"
        srcset="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=256&q=75 256w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=384&q=75 384w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=640&q=75 640w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=750&q=75 750w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=828&q=75 828w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1080&q=75 1080w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1200&q=75 1200w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1920&q=75 1920w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=2048&q=75 2048w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <div
      class="h-full flex flex-col gap-2 px-2 pb-2"
    >
      <p
        class="max-h-[calc(1.5rem_*_3)] line-clamp-3 font-extrabold text-blog-foreground-readable text-base transition-all duration-500"
        data-testid="article-card-title"
      >
        Suppellex a cognatus arca aliquam.
      </p>
      <p
        class="max-h-[calc(1.25rem_*_3)] line-clamp-3 text-[hsl(0,0%,55%)] text-sm group-hover:underline"
        data-testid="article-card-description"
      >
        Adnuo natus arca odit subseco ambulo. Suasoria cupio admiratio facilis sonitus dolorum. Occaecati venio apto apud timor cubicularis asperiores vestigium conqueror tantillus.
      </p>
      <p
        class="mt-auto text-xs"
        data-testid="article-card-datetime"
      >
        <small>
          <time>
            10 de março de 2025, às 20:11
          </time>
        </small>
      </p>
    </div>
  </a>
</DocumentFragment>
`;

exports[`ArticleCardList matches the snapshot 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="max-w-screen-2xl mx-auto px-4 flex items-center mb-10"
    data-testid="article-card-list"
    id="article-card-list"
  >
    <h2
      class="sr-only"
      id="article-list-title"
    >
      Lista de Artigos
    </h2>
    <ul
      aria-labelledby="article-list-title"
      class="w-full grid grid-cols-3 max-[800px]:grid-cols-2 max-[500px]:grid-cols-1 gap-4"
      role="list"
    >
      <li
        class="w-full max-h-[calc(200px_+_0.5rem_+_1.5rem_*_3_+_1.25rem_*_3_+_1rem_+_0.5rem_*_2_+_0.5rem)] p-1 rounded-lg transition-shadow duration-500 hover:shadow-blog-highlight group"
        role="listitem"
      >
        <a
          aria-label="Clique para ler mais sobre o artigo"
          class="border border-blog-border overflow-hidden h-full w-full flex flex-col rounded focus-visible:outline focus-visible:outline-2 focus-visible:outline-blog-foreground-readable-hover"
          data-testid="article-card-link"
          href="/artigos/bee102e1-6f8b-45ad-90cf-b4c17e4420a3/crux-fugit-curatio-stillicidium-bardus"
          id="article-card-link"
          role="link"
        >
          <div
            class="relative h-[200px] shrink-0 mb-2"
            data-testid="article-card-image"
          >
            <img
              alt="Approbo vicissitudo subiungo."
              class="absolute object-cover"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              role="img"
              sizes="(max-width: 1920) 100vw, (max-width: 960) 50vw, 33vw"
              src="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=3840&q=75"
              srcset="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=256&q=75 256w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=384&q=75 384w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=640&q=75 640w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=750&q=75 750w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=828&q=75 828w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1080&q=75 1080w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1200&q=75 1200w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1920&q=75 1920w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=2048&q=75 2048w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="h-full flex flex-col gap-2 px-2 pb-2"
          >
            <p
              class="max-h-[calc(1.5rem_*_3)] line-clamp-3 font-extrabold text-blog-foreground-readable text-base transition-all duration-500"
              data-testid="article-card-title"
            >
              Suppellex a cognatus arca aliquam.
            </p>
            <p
              class="max-h-[calc(1.25rem_*_3)] line-clamp-3 text-[hsl(0,0%,55%)] text-sm group-hover:underline"
              data-testid="article-card-description"
            >
              Adnuo natus arca odit subseco ambulo. Suasoria cupio admiratio facilis sonitus dolorum. Occaecati venio apto apud timor cubicularis asperiores vestigium conqueror tantillus.
            </p>
            <p
              class="mt-auto text-xs"
              data-testid="article-card-datetime"
            >
              <small>
                <time>
                  10 de março de 2025, às 20:11
                </time>
              </small>
            </p>
          </div>
        </a>
      </li>
      <li
        class="w-full max-h-[calc(200px_+_0.5rem_+_1.5rem_*_3_+_1.25rem_*_3_+_1rem_+_0.5rem_*_2_+_0.5rem)] p-1 rounded-lg transition-shadow duration-500 hover:shadow-blog-highlight group"
        role="listitem"
      >
        <a
          aria-label="Clique para ler mais sobre o artigo"
          class="border border-blog-border overflow-hidden h-full w-full flex flex-col rounded focus-visible:outline focus-visible:outline-2 focus-visible:outline-blog-foreground-readable-hover"
          data-testid="article-card-link"
          href="/artigos/eb10003d-88d1-449f-a80c-3c6db8201013/crux-fugit-curatio-stillicidium-bardus"
          id="article-card-link"
          role="link"
        >
          <div
            class="relative h-[200px] shrink-0 mb-2"
            data-testid="article-card-image"
          >
            <img
              alt="Virga iusto veritas saepe cribro depereo peccatus inflammatio vester versus crur vix."
              class="absolute object-cover"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              role="img"
              sizes="(max-width: 1920) 100vw, (max-width: 960) 50vw, 33vw"
              src="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=3840&q=75"
              srcset="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=256&q=75 256w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=384&q=75 384w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=640&q=75 640w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=750&q=75 750w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=828&q=75 828w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1080&q=75 1080w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1200&q=75 1200w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1920&q=75 1920w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=2048&q=75 2048w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="h-full flex flex-col gap-2 px-2 pb-2"
          >
            <p
              class="max-h-[calc(1.5rem_*_3)] line-clamp-3 font-extrabold text-blog-foreground-readable text-base transition-all duration-500"
              data-testid="article-card-title"
            >
              Suppellex a cognatus arca aliquam.
            </p>
            <p
              class="max-h-[calc(1.25rem_*_3)] line-clamp-3 text-[hsl(0,0%,55%)] text-sm group-hover:underline"
              data-testid="article-card-description"
            >
              Abundans aequitas volaticus minus barba casus tam. Nobis voluntarius tyrannus capto. Quam tricesimus aro accusamus aegrus depereo.
            </p>
            <p
              class="mt-auto text-xs"
              data-testid="article-card-datetime"
            >
              <small>
                <time>
                  10 de março de 2025, às 20:11
                </time>
              </small>
            </p>
          </div>
        </a>
      </li>
      <li
        class="w-full max-h-[calc(200px_+_0.5rem_+_1.5rem_*_3_+_1.25rem_*_3_+_1rem_+_0.5rem_*_2_+_0.5rem)] p-1 rounded-lg transition-shadow duration-500 hover:shadow-blog-highlight group"
        role="listitem"
      >
        <a
          aria-label="Clique para ler mais sobre o artigo"
          class="border border-blog-border overflow-hidden h-full w-full flex flex-col rounded focus-visible:outline focus-visible:outline-2 focus-visible:outline-blog-foreground-readable-hover"
          data-testid="article-card-link"
          href="/artigos/fd0252c5-f9ed-4e78-9c47-9096c5e92c90/crux-fugit-curatio-stillicidium-bardus"
          id="article-card-link"
          role="link"
        >
          <div
            class="relative h-[200px] shrink-0 mb-2"
            data-testid="article-card-image"
          >
            <img
              alt="Caelestis omnis color veritatis verbera."
              class="absolute object-cover"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              role="img"
              sizes="(max-width: 1920) 100vw, (max-width: 960) 50vw, 33vw"
              src="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=3840&q=75"
              srcset="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=256&q=75 256w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=384&q=75 384w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=640&q=75 640w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=750&q=75 750w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=828&q=75 828w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1080&q=75 1080w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1200&q=75 1200w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1920&q=75 1920w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=2048&q=75 2048w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="h-full flex flex-col gap-2 px-2 pb-2"
          >
            <p
              class="max-h-[calc(1.5rem_*_3)] line-clamp-3 font-extrabold text-blog-foreground-readable text-base transition-all duration-500"
              data-testid="article-card-title"
            >
              Suppellex a cognatus arca aliquam.
            </p>
            <p
              class="max-h-[calc(1.25rem_*_3)] line-clamp-3 text-[hsl(0,0%,55%)] text-sm group-hover:underline"
              data-testid="article-card-description"
            >
              Versus a voluptatibus cotidie voluptate pecco tricesimus necessitatibus. Circumvenio ocer tandem una temeritas sublime urbs commodi. Defendo creber cunctatio cuius cometes quasi curtus voluptate.
            </p>
            <p
              class="mt-auto text-xs"
              data-testid="article-card-datetime"
            >
              <small>
                <time>
                  10 de março de 2025, às 20:11
                </time>
              </small>
            </p>
          </div>
        </a>
      </li>
      <li
        class="w-full max-h-[calc(200px_+_0.5rem_+_1.5rem_*_3_+_1.25rem_*_3_+_1rem_+_0.5rem_*_2_+_0.5rem)] p-1 rounded-lg transition-shadow duration-500 hover:shadow-blog-highlight group"
        role="listitem"
      >
        <a
          aria-label="Clique para ler mais sobre o artigo"
          class="border border-blog-border overflow-hidden h-full w-full flex flex-col rounded focus-visible:outline focus-visible:outline-2 focus-visible:outline-blog-foreground-readable-hover"
          data-testid="article-card-link"
          href="/artigos/d6931872-98a2-4c38-8c05-dd8dfdd1ab9c/crux-fugit-curatio-stillicidium-bardus"
          id="article-card-link"
          role="link"
        >
          <div
            class="relative h-[200px] shrink-0 mb-2"
            data-testid="article-card-image"
          >
            <img
              alt="Curto."
              class="absolute object-cover"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              role="img"
              sizes="(max-width: 1920) 100vw, (max-width: 960) 50vw, 33vw"
              src="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=3840&q=75"
              srcset="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=256&q=75 256w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=384&q=75 384w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=640&q=75 640w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=750&q=75 750w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=828&q=75 828w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1080&q=75 1080w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1200&q=75 1200w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=1920&q=75 1920w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=2048&q=75 2048w, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fusr%2Fshare%2Fwaft.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="h-full flex flex-col gap-2 px-2 pb-2"
          >
            <p
              class="max-h-[calc(1.5rem_*_3)] line-clamp-3 font-extrabold text-blog-foreground-readable text-base transition-all duration-500"
              data-testid="article-card-title"
            >
              Suppellex a cognatus arca aliquam.
            </p>
            <p
              class="max-h-[calc(1.25rem_*_3)] line-clamp-3 text-[hsl(0,0%,55%)] text-sm group-hover:underline"
              data-testid="article-card-description"
            >
              Suus caveo cauda ratione conitor theca decumbo tero vulariter. Aranea venia id voluptatem repellat. In est appositus constans acquiro arto tamdiu adaugeo cornu unus.
            </p>
            <p
              class="mt-auto text-xs"
              data-testid="article-card-datetime"
            >
              <small>
                <time>
                  10 de março de 2025, às 20:11
                </time>
              </small>
            </p>
          </div>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
